lib/elixir/lib/enum.ex

```diff
@@ -4469,29 +4469,15 @@ defmodule Enum do
 
       {:error, module} ->
         {list, count} =
-          if step == 1 do
-            enumerable
-            |> module.reduce({:cont, {[], 0}}, fn elem, {acc, count} ->
-              {:cont, {[elem | acc], count + 1}}
-            end)
-            |> elem(1)
-          else
-            # We want to count all elements but we only keep the ones we need
-            {_, {list, _, count}} =
```

Comment:
I wonder if we can keep this optimization in some cases, for instance if we know the `start` to be positive (but currently we are not passing `start` at this stage).